python/lsst/daf/butler/core/storageClass.py

```diff
@@ -169,6 +181,80 @@ def _lookupNames(self):
         """
         return (self.name, )
 
+    def knownParameters(self):
+        """Return set of all parameters known to this `StorageClass`
+
+        The set includes parameters understood by components of a composite.
```

Comment:
We could probably get away with requiring the parameters of a composite to be a superset of the parameters of its components, and hence make `knownParameters` always the same as `parameters`.  At least I can't come up with a kind of parameter for which that would not be true with the StorageClasses and parameters I envision.

Reply:
For virtual composites the current behavior is exactly what you'd want (components can have their own parameters distinct from the parent).  Given the constraint that concrete composite assemblers have to be able to deal with all the parameters you might be right.  This automated collecting of child parameters seems like the right approach in general though.